### PR TITLE
Remove public packet fields

### PR DESCRIPTION
**Pull request by itmecho**
Removes public access to struct fields on the following items:
- [x] src/packet/ack.rs
- [x] src/packet/error.rs
- [x] src/packet/mod.rs
- [x] src/packet/data.rs
- [x] src/packet/rq/wrq.rs
- [x] src/packet/rq/mod.rs
- [x] src/packet/rq/rrq.rs

### CODE REVIEW

File: src/packet/data.rs [hunk 1]
```diff
@@ -29,6 +29,16 @@ impl Data {
...
         }
     }
+
+    /// Returns the identifier for the data
+    pub fn block(&self) -> Block {
+        self.block
+    }
+
+    /// Returns the payload
+    pub fn data(&self) -> &Vec<u8> {
```
Comment (itmecho) [inline src/packet/data.rs:39]: This feels a little gross. It might be better to switch this for implementing `AsRef<[u8]>`?

**Comment by connorkuehl** [comment]
I'm really sorry for the toil, @itmecho, but after reading a bit about some more Rust idioms I'm not sure we should add accessor methods. It seems the general consensus would be to add them to protect invariants or if certain combinations of fields don't make sense. However, we don't really have any invariants to protect here and I don't think the `packet` module should impose any additional policies about the data it manages.

What are your thoughts?

**Comment by itmecho** [comment]
No worries! Do you have any links so I can read up on it as well?

**Comment by connorkuehl** [comment]
Yeah! The main resource I've been reading is the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/future-proofing.html#c-struct-private) and also seeing what others thing on [Reddit posts like this one](https://www.reddit.com/r/rust/comments/4bs50v/are_accessors_idiomatic_in_rust/).

Sounds good. I will go ahead and close the issue I opened.

Thanks again (and sorry for the work)!

**Comment by connorkuehl** [comment]
Closing per discussion above.

**Comment by itmecho** [comment]
@connorkuehl Thanks for those links! Super useful =]

**Pull request closed without merging**